quinn-proto/src/connection.rs

```diff
@@ -834,20 +831,20 @@ impl Connection {
 
     fn init_0rtt(&mut self) {
         if self.side.is_client() && self.tls.get_early_secret().is_some() {
-            let params = self
-                .tls
-                .get_quic_transport_parameters()
-                .expect("rustls didn't supply transport parameters with ticket");
-            if let Err(e) = TransportParameters::read(self.side, &mut io::Cursor::new(params))
-                .map_err(Into::into)
-                .and_then(|x| self.set_params(x))
-            {
-                error!(
-                    self.log,
-                    "session ticket had malformed transport parameters: {}", e
-                );
-                return;
-            }
+            self.tls
+                .transport_parameters()
+                .and_then(|params| {
+                    self.set_params(
+                        params.expect("rustls didn't supply transport parameters with ticket"),
+                    )
+                    .map_err(Into::into)
+                })
+                .unwrap_or_else(|e| {
+                    error!(
+                        self.log,
+                        "session ticket has malformed transport parameters: {}", e
+                    )
+                })
```

Comment:
Accidentally removed `return`? We need to bail out of the function here if the transport parameters are missing or malformed, because sending 0-RTT in that case is forbidden.

Reply:
Fixed.